contextualized/regression/metamodels.py

```diff
@@ -22,12 +22,10 @@ def __init__(
         x_dim,
         y_dim,
         univariate=False,
-        encoder_type="mlp",
-        encoder_kwargs={
-            "width": 25,
-            "layers": 1,
-            "link_fn": LINK_FUNCTIONS["identity"],
-        },
+        encoder_type: str = "mlp",
+        width: int = 25,
+        layers: int = 1,
+        link_fn: callable = LINK_FUNCTIONS["identity"],
```

Comment:
Good job expanding this, other metamodels should also be expanded. Some metamodel signatures are different as well. 

I think it would be fine for now to expose consistent args and num_archetypes. For one-off args like in the tasksplit metamodel, you can still allow these to be passed using **kwargs. 
https://github.com/cnellington/Contextualized/blob/956b5dec3322bf4e6655d0464828874d76aa291e/contextualized/regression/metamodels.py#L199-L219